File: 05 - Matemática/1212 - Aritmetica Primaria/problem1212.cpp
#include <iostream>
 
using namespace std;

int calcularCarry(unsigned long x, unsigned long y);
void imprimirResposta(int carry);

int main() {
	unsigned long x, y;
	int carry;
	
	while (1) {
		
		cin >> x >> y;		

		if (x == 0  && y == 0) {
			break;
		}
		
		carry = calcularCarry(x, y);
		
		imprimirResposta(carry);	
	} 


    return 0;
}

int calcularCarry(unsigned long x, unsigned long y) {
	unsigned long rx, ry;
	int carry = 0, sumCarry = 0;

	while (x || y) {
		rx = x%10;
		ry = y%10;
		
		if ((rx + ry + carry) > 9) {
			sumCarry++;
			carry = 1;
		} else {
			carry = 0;
		}	
		
		x /= 10;
		y /= 10;
		
	} 
	
	return sumCarry;
}

void imprimirResposta(int carry) {		
	if (carry == 0) {
		cout << "No carry operations." << endl;
	} else if (carry == 1) {
		cout << "1 carry operation." << endl;		
	} else {
		cout << carry << " carry operations." << endl;		
	}
}

File: 05 - Matemática/1212 - Aritmetica Primaria/problem1212.py

def calcularCarry(x,  y):
	sumCarry = 0
	carry = 0

	while (x or y): 
		rx = x % 10
		ry = y % 10
		
		if ((rx + ry + carry) > 9): 
			sumCarry += 1
			carry = 1
		else: 
			carry = 0
				
		x /= 10
		y /= 10
	
	
	return sumCarry


def imprimirResposta(carry): 		
	if (carry == 0): 
		print("No carry operations.")
	elif (carry == 1):
		print("1 carry operation.")		
	else:
		print("%i carry operations." % carry)			


while True:
    
    x, y = input().split()		

    if (int(x) == 0  and int(y) == 0): 
        break;
    
    
    carry = calcularCarry(int(x), int(y))
    
    imprimirResposta(carry)		

